packages/opentelemetry-resources/src/Resource.ts

```diff
@@ -66,14 +113,30 @@ export class Resource {
    * @returns the newly merged Resource.
    */
   merge(other: Resource | null): Resource {
-    if (!other || !Object.keys(other.attributes).length) return this;
+    if (!other) return this;
 
     // SpanAttributes from resource overwrite attributes from other resource.
     const mergedAttributes = Object.assign(
       {},
       this.attributes,
       other.attributes
     );
-    return new Resource(mergedAttributes);
+
+    let mergedAsyncAttributesPromise: Promise<ResourceAttributes> | undefined;
```

Comment:
Worth checking if async promises are resolved and adding them to the sync attributes? Probably not very likely since resources will generally all be constructed almost at the same time.